### ISSUE TITLE
huge size of compiled executable

### ISSUE THREAD

**Issue by zhao-huang** [koka-lang/koka#277]
Hello there!
When I tried Koka, I found that the binaries generated by the compiler were very large, sometimes up to 15 MB.
For example, the following code generates a binary file of up to 2200 KB:
``` koka
import std/time
import std/num/random

pub fun main()
    local-time-now().show.println
    random-int().show.println
```
(Another issue is that the code above does not compile when targeting WASM on Windows, I haven't compiled it on Linux yet.)
I wonder if I am missing any compilation options, or simply Koka is currently not optimised for this?

**Comment by anfelor** [comment]
Yes, Koka is not really optimized for this yet and by default includes the entire standard library into all binaries. However, it compiles down to C and so you could try to remove the dead code at that phase:

Original size: 2321777
... after calling `strip`: 1714984
Size when compiling with `--ccopts="-ffunction-sections -fdata-sections" --cclinkopts="-Wl,-dead_strip"`: 830721
... and calling `strip`: 594104
Size with `dead_strip` and using `--fstdalloc` to exclude mimalloc: 765041
... and calling strip: 542536

(`-dead_strip` is for Macs, I believe that `--gc-sections` would work on other platforms).

The binary still seemed to work well after this transformation, but we do not actively support that process (and especially `-dead-strip` can sometimes remove necessary code fragments).
